cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/HashLoginServiceApiCredentials.java

```diff
@@ -0,0 +1,333 @@
+                        "HashLoginService's file format `username: password [,rolename ...]`");
+            }
+        });
+        return entries;
+    }
+
+    /**
+     * Returns Cruise Control API auth credentials file as a String.
+     *
+     * @param entries Map of API user entries containing API user credentials.
+     *
+     * @return Returns Cruise Control API auth credentials file as a String.
+     */
+    private static String generateApiAuthFileAsString(Map<String, UserEntry> entries) {
+        StringBuilder sb = new StringBuilder();
+        // Follows  Jetty's HashLoginService's file format: username: password [,rolename ...]
+        for (UserEntry e : entries.values()) {
+            sb.append(e.username).append(": ").append(e.password).append(", ").append(e.role()).append("\n");
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Parses auth data from existing Topic Operator API user secret into map of API user entries.
+     *
+     * @param secret API user secret
+     *
+     * @return Map of API user entries containing user-managed API user credentials
+     */
+    /* test */ static Map<String, UserEntry> generateTOManagedApiCredentials(Secret secret) {
+        Map<String, UserEntry> entries = new HashMap<>();
+        if (secret != null) {
+            if (secret.getData().containsKey(TOPIC_OPERATOR_USERNAME_KEY) && secret.getData().containsKey(TOPIC_OPERATOR_PASSWORD_KEY)) {
+                String username = Util.decodeFromBase64(secret.getData().get(TOPIC_OPERATOR_USERNAME_KEY));
+                String password = Util.decodeFromBase64(secret.getData().get(TOPIC_OPERATOR_PASSWORD_KEY));
+                entries.put(username, new UserEntry(username, password, Role.ADMIN));
+            }
+        }
+        return entries;
+    }
+
+    /**
+     * Parses auth data from existing user-managed API user secret into List of API user entries.
+     *
+     * @param secret API user secret
+     * @param secretKey API user secret key
+     *
+     * @return Map of API user entries containing user-managed API user credentials
+     */
+    /* test */ static Map<String, UserEntry> generateUserManagedApiCredentials(Secret secret, String secretKey) {
+        Map<String, UserEntry> entries = new HashMap<>();
+        if (secret != null) {
+            if (secretKey != null && secret.getData().containsKey(secretKey)) {
+                String credentialsAsString = Util.decodeFromBase64(secret.getData().get(secretKey));
+                entries.putAll(parseEntriesFromString(credentialsAsString));
+            }
+        }
+        for (UserEntry entry : entries.values()) {
+            if (FORBIDDEN_USERNAMES.contains(entry.username())) {
+                throw new InvalidConfigurationException("The following usernames for Cruise Control API are forbidden: " + FORBIDDEN_USERNAMES
+                        + " User provided Cruise Control API credentials contain illegal username: " + entry.username);
```

Comment:
This will not have any new line in the actual logs. So you should probably separate it somehow in the text as two different sentences.

```suggestion
                throw new InvalidConfigurationException("The following usernames for Cruise Control API are forbidden: " + FORBIDDEN_USERNAMES
                        + ". User provided Cruise Control API credentials contain illegal username: " + entry.username + ".");
```